core/modules/wiki.js

```diff
@@ -1035,6 +1035,18 @@ Options include:
 exports.parseText = function(type,text,options) {
 	text = text || "";
 	options = options || {};
+	var Parser = this.getParser(type,options)
+	// Return the parser instance
+	return new Parser(type,text,{
+		parseAsInline: options.parseAsInline,
+		wiki: this,
+		_canonical_uri: options._canonical_uri,
+		configTrimWhiteSpace: options.configTrimWhiteSpace
+	});
+};
+
+exports.getParser = function(type,options) {
+	options = $tw.utils.extend({},options);
```

Comment:
IMO your code will throw an RSOD if options is undefined
So `options = options || {}` should be used instead. We use this pattern all over the places.